lib/game.rb

```diff
@@ -0,0 +1,34 @@
+# rubocop:disable Metrics/AbcSize,Metrics/CyclomaticComplexity,Metrics/MethodLength,Metrics/PerceivedComplexity
+class Game
+  def winning_condition(arr, player_name)
+    if (arr[0] == arr[1] && arr[1] == arr[2]) && arr[0..2].none?('_')
+      puts "#{player_name} won on first line"
+      true
+    elsif (arr[3] == arr[4] && arr[4] == arr[5]) && arr[3..5].none?('_')
+      puts "#{player_name} won on the second line"
+      true
+    elsif (arr[6] == arr[7] && arr[7] == arr[8]) && arr[6..8].none?('_')
+      puts "#{player_name} won on the third line"
+      true
+    elsif (arr[0] == arr[3] && arr[3] == arr[6]) && !arr.values_at(0, 3, 6).include?('_')
```

Comment:
- [ ]  this not an acceptable way for checking for the winners
You can define methods to handle that:
```ruby
def check_win(player_symbol)
    # for rows
    all_equal(board[0..2], player_symbol) ||
    all_equal(board[3..5], player_symbol) ||
    all_equal(board[6..8], player_symbol)
  end

  def all_equal(subarray, player_symbol)
    return true if subarray.uniq.size == 1 && subarray.uniq == player_symbol
  end
```

Reply:
I really don't understand you tho, is there a fixed method to check for winning condition?.. please kindly explain. I already remove puts from the game class and it is working perfectly.

Reply:
@Eshy10 I am not trying to imply that my suggestion is the only way that works. Surely your way also does the job but it isn't the better way to handle that logic. You have a lot of if/else blocks in your code. I have only advised an alternative that can make your code cleaner.